frontend/src/api/k8s/templates.ts

```diff
@@ -39,18 +47,35 @@ export const assembleServingRuntimeTemplate = (
   };
 };
 
-export const listTemplates = async (
-  namespace?: string,
-  labelSelector?: string,
-): Promise<TemplateKind[]> => {
-  const queryOptions = {
-    ...(namespace && { ns: namespace }),
-    ...(labelSelector && { queryParams: { labelSelector } }),
-  };
-  return k8sListResource<TemplateKind>({
-    model: TemplateModel,
-    queryOptions,
-  }).then((listResource) => listResource.items);
+export const useTemplates = (namespace?: string): WatchK8sResult<TemplateKind[]> => {
+  const modelServingEnabled = useModelServingEnabled();
+  const customServingRuntimesEnabled = useCustomServingRuntimesEnabled();
+
+  if (!namespace) {
+    throw new NotReadyError('No namespace provided');
+  }
+
+  if (!modelServingEnabled) {
+    throw new NotReadyError('Model serving is not enabled');
+  }
+
+  const [templates, loaded, error] = useK8sWatchResource<TemplateKind[]>(
+    {
+      isList: true,
+      groupVersionKind: groupVersionKind(TemplateModel),
+      namespace,
+    },
+    TemplateModel,
+  );
+
+  if (!customServingRuntimesEnabled) {
+    return [
+      templates.filter((template) => template.metadata.labels?.['opendatahub.io/ootb'] === 'true'),
+      loaded,
+      error,
+    ];
+  }
+  return [templates, loaded, error];
```

Comment:
Store the filtered templates with `useMemo` and return a stable templates array.

Something like this:
```suggestion
  const t = React.useMemo(() => customServingRuntimesEnabled ? templates : templates.filter((template) => template.metadata.labels?.['opendatahub.io/ootb'] === 'true'), [templates, customServingRuntimesEnabled]);
  return [t, loaded, error];
```